### ISSUE TITLE
iOS 14 / Xcode12 / SwiftUI -> Quick Start

### ISSUE THREAD

**Issue by mcm-gr** [spotify/ios-sdk#312]
Hi there, I'm just getting started with figuring out how to integrate Spotify API into iOS 14 / Swift UI. However, I especially had a few troubles with Obj-C Linker Flag and AppDelegate which isn't available in current iOS anymore. 
 
Is there any **resource or sample project on a quick start** with the current **iOS 14** / XCode 12?
Thank you 
Best, Marlon

**Comment by mcm-gr** [comment]
Anyone with an idea?

**Comment by SamyTD** [comment]
I have the same problem

**Comment by tillhainbach** [comment]
here your go:
https://github.com/tillhainbach/SpotifyQuickStart

**Comment by downdrown** [comment]
@tillhainbach Thank you for the sample project, works like a charm. Could you eventually create a branch where you use the `SPTSessionManager#initiateSession` method instead of the `SPTAppRemote.authorizeAndPlayURI`?

**Comment by ir-fuel** [comment]
@downdrown  that requires the use of a backend + it's worthless anyway to do it like that because of a bunch of missing features in the SDK

**Comment by downdrown** [comment]
@ir-fuel I would not have a problem to provide a proper backend. I just don't want to have the user to re-authorize the app everytime he/she uses it. Is there a way to do this without the token swap backend? If so are there any good examples?
I am pretty new to iOS Development but I must say that the Spotify API feels a bit outdated and the documentation doesn't always seem to give me the anwers I'm looking for 😢

**Comment by ir-fuel** [comment]
@downdrown 

Read this and you'll understand why you should not waste your time (like I have) in building a backend for this.

https://github.com/spotify/ios-sdk/issues/343#issuecomment-1004853819

**Comment by ir-fuel** [comment]
No. You'll still see a ping-pong between your app and the Spotify app, but without a popup.

**Comment by ir-fuel** [comment]
And it's even worse than this. This SDK is worthless except for doing play/pause.

Imagine you start your app, Spotify is not playing, and you want the user to select a playlist for playback.
Well for one, there is no such thing as a "playlist". Spotify doesn't want you to navigate your own stuff in a structured way. The non-deprecated call in the API is a list of "recommendations" as what they think suits you.

But that's not the worst. The worst is you can't even call this method without Spotify actually PLAYING A SONG. The excuse you'll find in this Github is that "it is Apple's fault" because they can't be sure the Spotify app is open in the background and the only way to be sure is to have a song playing.
Well last time I checked Spotify also runs as a web player and there is absolutely no reason (except not willing to do it) to not pass these calls through a webservice to send back those lists (and hey, we have an access token thanks to that broken SessionManager). As long as you are not requesting actual playback it should work without going through the app.

The more I use this SDK the more I get frustrated with what a piece of badly documented junk it is. Incredible for a company of this size. Pretty sure there must be a few iOS developers @ HQ feeling frustrated because management doesn't let them make a decent SDK.

**Comment by DamilolaDami** [comment]
@ir-fuel is there any updates on this because its looking like its still very limited, although the nike run club app has integrated a very rich in app spotify experience. Not too sure how they achieved it

**Comment by ir-fuel** [comment]
No idea. Haven't been on this project for a long time.

On 13 Nov 2023, at 16:44, Dami ***@***.***> wrote:

﻿

@ir-fuel<https://github.com/ir-fuel> is there any updates on this because its looking like its still very limited, although the nike run club app has integrated a very rich in app spotify experience. Not too sure how they achieved it
